src/order/initial_seg.lean

```diff
@@ -312,6 +313,21 @@ def of_element {α : Type*} (r : α → α → Prop) (a : α) : subrel r {b | r
 @[simp] theorem of_element_top {α : Type*} (r : α → α → Prop) (a : α) :
   (of_element r a).top = a := rfl
 
+/-- Any principal segment of a well order is order-isomorphic to a `subrel` -/
+@[simps apply]
+noncomputable def iso_subrel (f : r ≺i s) : r ≃r subrel s {b | s b f.top} :=
```

Comment:
The existence of `ordinal.enum_iso` suggests that this is the wrong way around. Can you make this one
```suggestion
noncomputable def iso_subrel (f : r ≺i s) : subrel s {b | s b f.top}≃r r :=
```
or turn `ordinal.enum_iso` around?